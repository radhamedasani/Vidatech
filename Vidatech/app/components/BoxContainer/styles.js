import { StyleSheet } from 'react-native'
import { padding, colors } from '../../themes/base'

export const boxStyles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        marginBottom: padding.sm,
        marginHorizontal: padding.sm,
        paddingLeft: padding.sm,
        backgroundColor: colors.secondary
    }
});