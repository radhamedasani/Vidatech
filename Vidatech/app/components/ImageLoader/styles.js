import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
    loader: {
        position: 'absolute',
        alignItems: 'center',
        justifyContent: 'center'
    }
})