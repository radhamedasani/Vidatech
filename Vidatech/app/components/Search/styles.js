import { StyleSheet } from 'react-native'
import { colors, padding, fontSize } from '../../themes/base'
export const searchStyle = StyleSheet.create({
    container: {
        flexDirection: 'row',
        borderColor: 'grey',
        borderWidth: 1,
        borderRadius: 20,
        height: 50,
        backgroundColor: colors.secondary,
        marginBottom: 10,
        marginHorizontal: padding.sm,
        alignItems: 'center',
        paddingHorizontal: padding.sm
    },
    txtInput: {
        width: '90%',
        color: '#000',
        height:'100%',
      //  backgroundColor:'red',
        fontSize: fontSize.md,
        marginLeft: padding.sm
    },
    searchImg: {
        width: 25,
        height: 25
    }
})