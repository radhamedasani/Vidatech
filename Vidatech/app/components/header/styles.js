import { StyleSheet } from 'react-native'
import { colors } from '../../themes/base'
const styles = StyleSheet.create({
    container: {
        backgroundColor: colors.primary,
        alignItems: 'center',
        justifyContent: 'center',
        height: 60
    },
    title: {
        color: colors.secondary,
        fontSize: 24
    }
});
export default styles;