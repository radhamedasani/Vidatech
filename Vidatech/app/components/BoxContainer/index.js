import React from 'react'
import { View } from 'react-native'
import { boxStyles } from './styles'

export const BoxContainer = (props) => (
    <View style={boxStyles.container}>
        {props.children}
    </View>
)