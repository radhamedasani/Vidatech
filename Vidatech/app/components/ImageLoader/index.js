import React, { Component, useState, useEffect } from 'react'
import { View, Image, ActivityIndicator } from 'react-native'
import { styles } from './styles'

export function ImageLoader(props) {
    const { avatar, avatarStyle } = props;
    const { loader } = styles;
    const [loading, setLoading] = useState(true);
    const [srcImage, setSrcImge] = useState({ uri: avatar });
    const errorImg = require('./Images/error.jpg');
    const onDownloadImage = () => {
        setLoading(false);
    }
    const onError = () => {
        setLoading(false);
        setSrcImge(errorImg);
    }
    return (<View style={avatarStyle}>
        <View style={[loader, avatarStyle]}>
            {loading ? <ActivityIndicator animating={true} size='large' /> : null}
        </View>
        <View style={[loader, avatarStyle]}>
            <Image source={srcImage} style={avatarStyle}
                onLoad={onDownloadImage} onError={onError} />
        </View>
    </View>);
}   