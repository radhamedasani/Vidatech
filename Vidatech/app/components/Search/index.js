import React, { useState } from 'react'
import { View, TextInput, Image, TouchableWithoutFeedback } from 'react-native'
import { searchStyle } from './styles'

export const SearchBar = (props) => {
    const SEARCH_IMG = require('./Images/search.png');
    const [searchTxt, setSearchTxt] = useState('');
    const { placeholderTxt, onClickSearch } = props;
    const { container, txtInput, searchImg } = searchStyle;
    const onTextInput = (changedText) => {
       // alert(changedText);
        setSearchTxt(changedText);
        onClickSearch(changedText);
    }
    const onClickSearchButton = () => {
         onClickSearch(searchTxt);
    }
    return (<View style={container}>
        <TextInput style={txtInput}
            placeholder={placeholderTxt}
            placeholderTextColor='grey'
            value = {searchTxt}
            onChangeText={onTextInput} />
        <TouchableWithoutFeedback onPress={onClickSearchButton}>
            <Image style={searchImg}
                source={SEARCH_IMG}></Image>
        </TouchableWithoutFeedback>
    </View>);
}