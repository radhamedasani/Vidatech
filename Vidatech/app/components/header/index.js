import React from 'react'
import { View, Text } from 'react-native'
import styles from './styles'

export const Header = () => (
    <View style={styles.container}>
        <Text style={styles.title}>Vidatec</Text>
    </View>
)

