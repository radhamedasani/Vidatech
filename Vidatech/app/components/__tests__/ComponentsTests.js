import React from 'react'
import reactTestRender from 'react-test-renderer'
import { BoxContainer } from '../BoxContainer'
import { ImageLoader } from '../ImageLoader'
import { Header } from '../header'
import { SearchBar } from '../Search'

describe('Custom Components Tests ==>', () => {
    it('Test Box Container', () => {
        const boxComp = reactTestRender.create(<BoxContainer />);
        expect(boxComp).toMatchSnapshot();
    });
    it('Test Header Component', () => {
        const headerComp = reactTestRender.create(<Header />);
        expect(headerComp).toMatchSnapshot();
    });
    it('Test Image Loader', () => {
        const imageLoaderComp = reactTestRender.create(<ImageLoader />);
        expect(imageLoaderComp).toMatchSnapshot();
    });
    it('Test Search bar', () => {
        const searchComp = reactTestRender.create(<SearchBar placeholderTxt='Test People' />);
        expect(searchComp).toMatchSnapshot();
    });
})
