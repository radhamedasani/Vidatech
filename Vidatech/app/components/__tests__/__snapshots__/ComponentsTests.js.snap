// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Components Tests ==> Test Box Container 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 10,
      "marginHorizontal": 10,
      "paddingLeft": 10,
    }
  }
/>
`;

exports[`Custom Components Tests ==> Test Header Component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#882758",
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
      }
    }
  >
    Vidatec
  </Text>
</View>
`;

exports[`Custom Components Tests ==> Test Image Loader 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        undefined,
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        undefined,
      ]
    }
  >
    <Image
      onError={[Function]}
      onLoad={[Function]}
      source={
        Object {
          "uri": undefined,
        }
      }
    />
  </View>
</View>
`;

exports[`Custom Components Tests ==> Test Search bar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "grey",
      "borderRadius": 20,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "marginBottom": 10,
      "marginHorizontal": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Test People"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#000",
        "fontSize": 16,
        "height": "100%",
        "marginLeft": 10,
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../app/components/Search/Images/search.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
</View>
`;
